### ISSUE TITLE
Python : Flask Path Traversal Vulnerability

### ISSUE THREAD

**Issue by porcupineyhairs** [github/securitylab#669]
### CVE(s) ID list

This is a placeholder issue. I plan on sending bulk PR's to approx 100 projects. I will add the CVE once the  fixes are merged and identifiers are assigned. 

### All For One submission

https://github.com/github/securitylab/issues/407

### Details

TBA

### Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc).

- [X] Yes
- [ ] No

**Comment by jorgectf** [comment]
👋 Just a detail. There's a typo in `This bug was found using [CodeQL by Github](here)` pointing to affeted repo's `/issues/codeql.github.com` :)

**Comment by porcupineyhairs** [comment]
Hi,

So, all in all, I have been able to send in close to 92 reports. These are independent from the four CVE's referenced in #480. 

Here are the details : 

| Sr. No.| Project | Current State | CVE ID| Issue | Pull Request |
| :--| :-:| :-: | :-:| ---| ---|
|1|[onlaj/Piano-LED-Visualizer](https://github.com/onlaj/Piano-LED-Visualizer)|Advisory issued.|https://github.com/onlaj/Piano-LED-Visualizer/security/advisories/GHSA-g78x-q3x8-r6m4|https://github.com/onlaj/Piano-LED-Visualizer/issues/350|https://github.com/onlaj/Piano-LED-Visualizer/pull/351|
|2|[orchest/orchest](https://github.com/orchest/orchest)|Advisory issued.|https://github.com/orchest/orchest/security/advisories/GHSA-j2rj-x939-977h|https://github.com/orchest/orchest/issues/906|https://github.com/orchest/orchest/pull/907|
|3|[clinical-genomics/scout](https://github.com/clinical-genomics/scout)|Advisory issued.|https://nvd.nist.gov/vuln/detail/CVE-2022-1554|https://www.huntr.dev/bounties/7acac778-5ba4-4f02-99e2-e4e17a81e600/|https://www.github.com/clinical-genomics/scout/commit/952a2e2319af2d95d22b017a561730feac086ff1|
|4|[ChangeWeDer/BaiduWenkuSpider_flaskWeb](https://github.com/ChangeWeDer/BaiduWenkuSpider_flaskWeb)|PR Merged Advisory Pending.|||https://github.com/ChangeWeDer/BaiduWenkuSpider_flaskWeb/pull/3|
|5|[Microsoft/OLive](https://github.com/Microsoft/OLive)|PR Merged Advisory Pending.||https://www.huntr.dev/bounties/303b6e02-b51f-4f0b-96cf-f7f81da10fff/|https://github.com/microsoft/OLive/pull/83|
|6|[iedadata/usap-dc-website](https://github.com/iedadata/usap-dc-website)|PR Merged Advisory Pending.||https://www.huntr.dev/bounties/e22b9810-4cb4-4e33-8883-d9ccf1f64024/|iedadata/usap-dc-website@64f22e4, iedadata/usap-dc-website@afd39a1 and iedadata/usap-dc-website@d40f093|
|7|[operatorequals/wormnest](https://github.com/operatorequals/wormnest)|PR Merged Advisory Pending.||https://github.com/operatorequals/wormnest/issues/7|https://github.com/operatorequals/wormnest/pull/8|
|8|[sergeKashkin/Simple-RAT](https://github.com/sergeKashkin/Simple-RAT)|PR Merged Advisory Pending.||https://github.com/sergeKashkin/Simple-RAT/issues/10|https://github.com/sergeKashkin/Simple-RAT/pull/11|
|9|[AFDudley/equanimity](https://github.com/AFDudley/equanimity)|PR Review.||https://github.com/AFDudley/equanimity/issues/2|https://github.com/AFDudley/equanimity/pull/3|
|10|[Atom02/flask-mvc](https://github.com/Atom02/flask-mvc)|PR Review.||https://github.com/Atom02/flask-mvc/issues/1|https://github.com/Atom02/flask-mvc/pull/2|
|11|[BolunHan/Krypton](https://github.com/BolunHan/Krypton)|PR Review.||https://github.com/BolunHan/Krypton/issues/1|https://github.com/BolunHan/Krypton/pull/2|
|12|[Caoyongqi912/Fan_Platform](https://github.com/Caoyongqi912/Fan_Platform)|PR Review.||https://github.com/Caoyongqi912/Fan_Platform/issues/12|https://github.com/Caoyongqi912/Fan_Platform/pull/13|
|13|[ChaoticOnyx/OnyxForum](https://github.com/ChaoticOnyx/OnyxForum)|PR Review.||https://github.com/ChaoticOnyx/OnyxForum/issues/62|https://github.com/ChaoticOnyx/OnyxForum/pull/63|
|14|[Delor4/CarceresBE](https://github.com/Delor4/CarceresBE)|PR Review.||https://github.com/Delor4/CarceresBE/issues/3|https://github.com/Delor4/CarceresBE/pull/4|
|15|[Harveyzyh/Python](https://github.com/Harveyzyh/Python)|PR Review.||https://github.com/Harveyzyh/Python/issues/3|https://github.com/Harveyzyh/Python/pull/4|
|16|[HolgerGraef/MSM](https://github.com/HolgerGraef/MSM)|PR Review.||https://github.com/HolgerGraef/MSM/issues/129|https://github.com/HolgerGraef/MSM/pull/130|
|17|[JustAnotherSoftwareDeveloper/Python-Recipe-Database](https://github.com/JustAnotherSoftwareDeveloper/Python-Recipe-Database)|PR Review.||https://github.com/JustAnotherSoftwareDeveloper/Python-Recipe-Database/issues/9|https://github.com/JustAnotherSoftwareDeveloper/Python-Recipe-Database/pull/10|
|18|[Lukasavicus/WindMill](https://github.com/Lukasavicus/WindMill)|PR Review.||https://github.com/Lukasavicus/WindMill/issues/3|https://github.com/Lukasavicus/WindMill/pull/4|
|19|[Luxas98/logstash-management-api](https://github.com/Luxas98/logstash-management-api)|PR Review.||https://github.com/Luxas98/logstash-management-api/issues/3|https://github.com/Luxas98/logstash-management-api/pull/4|
|20|[Niyaz-Mohamed/mosaic](https://github.com/Niyaz-Mohamed/mosaic)|PR Review.|||https://github.com/Niyaz-Mohamed/mosaic/pull/18|
|21|[NotVinay/karaokey](https://github.com/NotVinay/karaokey)|PR Review.||https://github.com/NotVinay/karaokey/issues/5|https://github.com/NotVinay/karaokey/pull/6|
|22|[PaddlePaddle/Anakin](https://github.com/PaddlePaddle/Anakin)|PR Review.||https://github.com/PaddlePaddle/Anakin/issues/548|https://github.com/PaddlePaddle/Anakin/pull/549|
|23|[PureStorage-OpenConnect/swagger](https://github.com/PureStorage-OpenConnect/swagger)|PR Review.||https://github.com/PureStorage-OpenConnect/swagger/issues/9|https://github.com/PureStorage-OpenConnect/swagger/pull/10|
|24|[SummaLabs/DLS](https://github.com/SummaLabs/DLS)|PR Review.||https://github.com/SummaLabs/DLS/issues/4|https://github.com/SummaLabs/DLS/pull/5|
|25|[ThundeRatz/ThunderDocs](https://github.com/ThundeRatz/ThunderDocs)|PR Review.||https://github.com/ThundeRatz/ThunderDocs/issues/1|https://github.com/ThundeRatz/ThunderDocs/pull/2|
|26|[Wildog/flask-file-server](https://github.com/Wildog/flask-file-server)|PR Review.||https://github.com/Wildog/flask-file-server/issues/17|https://github.com/Wildog/flask-file-server/pull/18|
|27|[adobe/OSAS](https://github.com/adobe/OSAS)|PR Review.||https://github.com/adobe/OSAS/issues/16|https://github.com/adobe/OSAS/pull/17|
|28|[bonn-activity-maps/bam_annotation_tool](https://github.com/bonn-activity-maps/bam_annotation_tool)|PR Review.||https://github.com/bonn-activity-maps/bam_annotation_tool/issues/474|https://github.com/bonn-activity-maps/bam_annotation_tool/pull/476|
|29|[cheo0/MercadoEnLineaBack](https://github.com/cheo0/MercadoEnLineaBack)|PR Review.||https://github.com/cheo0/MercadoEnLineaBack/issues/1|https://github.com/cheo0/MercadoEnLineaBack/pull/2|
|30|[cinemaproject/monorepo](https://github.com/cinemaproject/monorepo)|PR Review.||https://github.com/cinemaproject/monorepo/issues/46|https://github.com/cinemaproject/monorepo/pull/47|
|31|[cmusatyalab/opendiamond](https://github.com/cmusatyalab/opendiamond)|PR Review.||https://github.com/cmusatyalab/opendiamond/issues/52|https://github.com/cmusatyalab/opendiamond/pull/53|
|32|[csm-aut/csm](https://github.com/csm-aut/csm)|PR Review.||https://github.com/csm-aut/csm/issues/104|https://github.com/csm-aut/csm/pull/105|
|33|[dainst/cilantro](https://github.com/dainst/cilantro)|PR Review.||https://github.com/dainst/cilantro/issues/260|https://github.com/dainst/cilantro/pull/261|
|34|[dankolbman/travel_blahg](https://github.com/dankolbman/travel_blahg)|PR Review.||https://github.com/dankolbman/travel_blahg/issues/3|https://github.com/dankolbman/travel_blahg/pull/4|
|35|[decentraminds/umbral](https://github.com/decentraminds/umbral)|PR Review.||https://github.com/decentraminds/umbral/issues/13|https://github.com/decentraminds/umbral/pull/14|
|36|[echoleegroup/PythonWeb](https://github.com/echoleegroup/PythonWeb)|PR Review.||https://github.com/echoleegroup/PythonWeb/issues/1|https://github.com/echoleegroup/PythonWeb/pull/2|
|37|[freefood89/Fishtank](https://github.com/freefood89/Fishtank)|PR Review.||https://github.com/freefood89/Fishtank/issues/4|https://github.com/freefood89/Fishtank/pull/5|
|38|[ganga-devs/ganga](https://github.com/ganga-devs/ganga)|PR Review.||https://github.com/ganga-devs/ganga/issues/2024|https://github.com/ganga-devs/ganga/pull/2025|
|39|[idayrus/evoting](https://github.com/idayrus/evoting)|PR Review.||https://github.com/idayrus/evoting/issues/1|https://github.com/idayrus/evoting/pull/2|
|40|[jaygarza1982/ytdl-sync](https://github.com/jaygarza1982/ytdl-sync)|PR Review.||https://github.com/jaygarza1982/ytdl-sync/issues/1|https://github.com/jaygarza1982/ytdl-sync/pull/2|
|41|[jmcginty15/Solar-system-simulator](https://github.com/jmcginty15/Solar-system-simulator)|PR Review.||https://github.com/jmcginty15/Solar-system-simulator/issues/1|https://github.com/jmcginty15/Solar-system-simulator/pull/2|
|42|[joaopedro-fg/mp-m08-interface](https://github.com/joaopedro-fg/mp-m08-interface)|PR Review.||https://github.com/joaopedro-fg/mp-m08-interface/issues/1|https://github.com/joaopedro-fg/mp-m08-interface/pull/2|
|43|[kotekan/kotekan](https://github.com/kotekan/kotekan)|PR Review.||https://github.com/kotekan/kotekan/issues/1024|https://github.com/kotekan/kotekan/pull/1025|
|44|[kumardeepak/hin-eng-preprocessing](https://github.com/kumardeepak/hin-eng-preprocessing)|PR Review.||https://github.com/kumardeepak/hin-eng-preprocessing/issues/32|https://github.com/kumardeepak/hin-eng-preprocessing/pull/33|
|45|[lyubolp/Barry-Voice-Assistant](https://github.com/lyubolp/Barry-Voice-Assistant)|PR Review.||https://github.com/lyubolp/Barry-Voice-Assistant/issues/57|https://github.com/lyubolp/Barry-Voice-Assistant/pull/58|
|46|[mandoku/mdweb](https://github.com/mandoku/mdweb)|PR Review.||https://github.com/mandoku/mdweb/issues/1|https://github.com/mandoku/mdweb/pull/2|
|47|[maxtortime/SetupBox](https://github.com/maxtortime/SetupBox)|PR Review.||https://github.com/maxtortime/SetupBox/issues/55|https://github.com/maxtortime/SetupBox/pull/56|
|48|[meerstein/rbtm](https://github.com/meerstein/rbtm)|PR Review.||https://github.com/meerstein/rbtm/issues/47|https://github.com/meerstein/rbtm/pull/48|
|49|[ml-inory/ModelConverter](https://github.com/ml-inory/ModelConverter)|PR Review.||https://github.com/ml-inory/ModelConverter/issues/1|https://github.com/ml-inory/ModelConverter/pull/2|
|50|[nlpweb/glance](https://github.com/nlpweb/glance)|PR Review.||https://github.com/nlpweb/glance/issues/7|https://github.com/nlpweb/glance/pull/3|
|51|[noamezekiel/sphere](https://github.com/noamezekiel/sphere)|PR Review.||https://github.com/noamezekiel/sphere/issues/19|https://github.com/noamezekiel/sphere/pull/18|
|52|[nrlakin/homepage](https://github.com/nrlakin/homepage)|PR Review.||https://github.com/nrlakin/homepage/issues/1|https://github.com/nrlakin/homepage/pull/2|
|53|[olmax99/helm-flask-celery](https://github.com/olmax99/helm-flask-celery)|PR Review.||https://github.com/olmax99/helm-flask-celery/issues/1|https://github.com/olmax99/helm-flask-celery/pull/2|
|54|[olmax99/pyathenastack](https://github.com/olmax99/pyathenastack)|PR Review.||https://github.com/olmax99/pyathenastack/issues/10|https://github.com/olmax99/pyathenastack/pull/11|
|55|[pleomax00/flask-mongo-skel](https://github.com/pleomax00/flask-mongo-skel)|PR Review.||https://github.com/pleomax00/flask-mongo-skel/issues/3|https://github.com/pleomax00/flask-mongo-skel/pull/2|
|56|[project-anuvaad/anuvaad-corpus](https://github.com/project-anuvaad/anuvaad-corpus)|PR Review.||https://github.com/project-anuvaad/anuvaad-corpus/issues/184|https://github.com/project-anuvaad/anuvaad-corpus/pull/185|
|57|[rainsoupah/sleep-learner](https://github.com/rainsoupah/sleep-learner)|PR Review.||https://github.com/rainsoupah/sleep-learner/issues/27|https://github.com/rainsoupah/sleep-learner/pull/28|
|58|[rohitnayak/movie-review-sentiment-analysis](https://github.com/rohitnayak/movie-review-sentiment-analysis)|PR Review.||https://github.com/rohitnayak/movie-review-sentiment-analysis/issues/1|https://github.com/rohitnayak/movie-review-sentiment-analysis/pull/2|
|59|[romain20100/nursequest](https://github.com/romain20100/nursequest)|PR Review.||https://github.com/romain20100/nursequest/issues/6|https://github.com/romain20100/nursequest/pull/7|
|60|[rusyasoft/TrainEnergyServer](https://github.com/rusyasoft/TrainEnergyServer)|PR Review.||https://github.com/rusyasoft/TrainEnergyServer/issues/2|https://github.com/rusyasoft/TrainEnergyServer/pull/3|
|61|[seveas/golem](https://github.com/seveas/golem)|PR Review.||https://github.com/seveas/golem/issues/11|https://github.com/seveas/golem/pull/12|
|62|[tooxie/shiva-server](https://github.com/tooxie/shiva-server)|PR Review.||https://github.com/tooxie/shiva-server/issues/189|https://github.com/tooxie/shiva-server/pull/190|
|63|[tsileo/flask-yeoman](https://github.com/tsileo/flask-yeoman)|PR Review.||https://github.com/tsileo/flask-yeoman/issues/2|https://github.com/tsileo/flask-yeoman/pull/3|
|64|[uncleYiba/photo_tag](https://github.com/uncleYiba/photo_tag)|PR Review.||https://github.com/uncleYiba/photo_tag/issues/1|https://github.com/uncleYiba/photo_tag/pull/2|
|65|[varijkapil13/Sphere_ImageBackend](https://github.com/varijkapil13/Sphere_ImageBackend)|PR Review.||https://github.com/varijkapil13/Sphere_ImageBackend/issues/23|https://github.com/varijkapil13/Sphere_ImageBackend/pull/24|
|66|[waveyan/internshipsystem](https://github.com/waveyan/internshipsystem)|PR Review.||https://github.com/waveyan/internshipsystem/issues/1|https://github.com/waveyan/internshipsystem/pull/2|
|67|[whmacmac/vprj](https://github.com/whmacmac/vprj)|PR Review.||https://github.com/whmacmac/vprj/issues/2|https://github.com/whmacmac/vprj/pull/3|
|68|[woduq1414/munhak-moa](https://github.com/woduq1414/munhak-moa)|PR Review.||https://github.com/woduq1414/munhak-moa/issues/1|https://github.com/woduq1414/munhak-moa/pull/2|
|69|[yogson/syrabond](https://github.com/yogson/syrabond)|PR Review.||https://github.com/yogson/syrabond/issues/1|https://github.com/yogson/syrabond/pull/2|
|70|[DSAB-local/DSAB](https://github.com/DSAB-local/DSAB)|Vulnerability reported.||https://github.com/DSAB-local/DSAB/issues/2||
|71|[DSABenchmark/DSAB](https://github.com/DSABenchmark/DSAB)|Vulnerability reported.||https://github.com/DSABenchmark/DSAB/issues/1||
|72|[Rexians/rex-web](https://github.com/Rexians/rex-web)|Vulnerability reported.||https://github.com/Rexians/rex-web/issues/5||
|73|[RipudamanKaushikDal/projects](https://github.com/RipudamanKaushikDal/projects)|Vulnerability reported.||https://github.com/RipudamanKaushikDal/projects/issues/21||
|74|[adriankoczuruek/ceneo-web-scrapper](https://github.com/adriankoczuruek/ceneo-web-scrapper)|Vulnerability reported.||https://www.huntr.dev/bounties/f23dd240-c93a-4ad0-ab25-e02ec2b1a2ae/||
|75|[akashtalole/python-flask-restful-api](https://github.com/akashtalole/python-flask-restful-api)|Vulnerability reported.||https://github.com/akashtalole/python-flask-restful-api/issues/2||
|76|[ceee-vip/cockybook](https://github.com/ceee-vip/cockybook)|Vulnerability reported.||https://github.com/ceee-vip/cockybook/issues/1||
|77|[chainer/chainerrl-visualizer](https://github.com/chainer/chainerrl-visualizer)|Vulnerability reported.||https://github.com/chainer/chainerrl-visualizer/issues/35||
|78|[deepaliupadhyay/RealEstate](https://github.com/deepaliupadhyay/RealEstate)|Vulnerability reported.||https://github.com/deepaliupadhyay/RealEstate/issues/4||
|79|[duducosmos/livro_python](https://github.com/duducosmos/livro_python)|Vulnerability reported.||https://github.com/duducosmos/livro_python/issues/1||
|80|[heidi-luong1109/shackerpanel](https://github.com/heidi-luong1109/shackerpanel)|Vulnerability reported.||https://github.com/heidi-luong1109/shackerpanel/issues/1||
|81|[longmaoteamtf/audio_aligner_app](https://github.com/longmaoteamtf/audio_aligner_app)|Vulnerability reported.||https://www.huntr.dev/bounties/b88bdc45-1b12-4b26-85c7-f5f56f3676fc/||
|82|[piaoyunsoft/bt_lnmp](https://github.com/piaoyunsoft/bt_lnmp)|Vulnerability reported.||https://github.com/piaoyunsoft/bt_lnmp/issues/7||
|83|[ralphjzhang/iasset](https://github.com/ralphjzhang/iasset)|Vulnerability reported.||https://github.com/ralphjzhang/iasset/issues/1||
|84|[sanojtharindu/caretakerr-api](https://github.com/sanojtharindu/caretakerr-api)|Vulnerability reported.||https://github.com/sanojtharindu/caretakerr-api/issues/1||
|85|[scorelab/OpenMF](https://github.com/scorelab/OpenMF)|Vulnerability reported.||https://github.com/scorelab/OpenMF/issues/263||
|86|[shaolo1/VideoServer](https://github.com/shaolo1/VideoServer)|Vulnerability reported.||https://github.com/shaolo1/VideoServer/issues/1||
|87|[sravaniboinepelli/AutomatedQuizEval](https://github.com/sravaniboinepelli/AutomatedQuizEval)|Vulnerability reported.||https://github.com/sravaniboinepelli/AutomatedQuizEval/issues/18||
|88|[stonethree/s3label](https://github.com/stonethree/s3label)|Vulnerability reported.||https://github.com/stonethree/s3label/issues/109||
|89|[umeshpatil-dev/Home__internet](https://github.com/umeshpatil-dev/Home__internet)|Vulnerability reported.||https://github.com/umeshpatil-dev/Home__internet/issues/1||
|90|[unizar-30226-2019-06/ChangePop-Back](https://github.com/unizar-30226-2019-06/ChangePop-Back)|Vulnerability reported.||https://github.com/unizar-30226-2019-06/ChangePop-Back/issues/14||
|91|[yuriyouzhou/KG-fashion-chatbot](https://github.com/yuriyouzhou/KG-fashion-chatbot)|Vulnerability reported.||https://github.com/yuriyouzhou/KG-fashion-chatbot/issues/2||
|92|[zippies/testplatform](https://github.com/zippies/testplatform)|Vulnerability reported.||https://www.huntr.dev/bounties/4f7aee47-8ca5-4511-8e13-6249be5ec005/||

**Comment by jorgectf** [comment]
:wave:  @porcupineyhairs

Impressive work, thank you for your submission!

> I will update this ticket as and when they are merged.

Feel free to ping me once you do :)

---

As per The Bug Slayer [rules](https://securitylab.github.com/bounties/#bugslayer), please consider the following:

> Please provide any information that proves that your query finds the CVE (for example a LGTM link, a CodeQL DB, a GitHub gist with the modified query)

Moreover, take into account that you might qualify for further bounties:

> Additionally, CVEs published as a result of your research into open source projects may be eligible for a bounty from the Internet Bug Bounty (IBB).

As per the CVSS, I beg to differ with `This can lead to a partial system denial service`, since being able to DoS by leaking large files means the connection hangs, but that depends on how `Flask` manages threads (production-ready servers use uwsgi/gunicorn with a very strong threading system), the server specifications, and ultimately the network speed. In other words, the vulnerability itself does not cause reduced performance or interruptions in resource availability, and if it does, it is not a problem originated by `Flask`.

Please see:

CVSS [relevant guidance](https://www.first.org/cvss/user-guide):

> If a specific configuration is required for an attack to succeed, the vulnerable component should be scored assuming it is in that configuration, providing it is a reasonable configuration.

And a reasonable/standard/realistic environment should be (regarding [flask documentation](https://flask.palletsprojects.com/en/2.1.x/tutorial/deploy/#run-with-a-production-server)):

> When running publicly rather than in development, you should not use the built-in development server (flask run). The development server is provided by Werkzeug for convenience, but is not designed to be particularly efficient, stable, or secure.

Indeed, All of the CVEs you specified [in this comment](https://github.com/github/securitylab/issues/407#issuecomment-976564884) were given an `Availability Impact` of `None`.

https://nvd.nist.gov/vuln/detail/CVE-2021-43492
https://nvd.nist.gov/vuln/detail/CVE-2021-43493
https://nvd.nist.gov/vuln/detail/CVE-2021-43494
https://nvd.nist.gov/vuln/detail/CVE-2021-43495
https://nvd.nist.gov/vuln/detail/CVE-2021-43496

[AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:N/A:N](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:N/A:N&version=3.1)

This leaves a base score of 8.6/10 and a severity rating of high.

**Comment by porcupineyhairs** [comment]
@jorgectf 

Many of the apps in the list I submit with this report recommend a user to run the app via flask's builtin web-server. Some even supply production scripts which end up calling `Flask`'s development web-server. Amongst the projects where the fix has been merged, here's a rough summary of how they recommend a user to deploy on production.

1. [onlaj/Piano-LED-Visualizer](https://github.com/onlaj/Piano-LED-Visualizer#web-interface) 
In this case, if the user follows the project docs thoroughly, they would run the app using flask's development web-server using the `sudo python3 /home/Piano-LED-Visualizer/visualizer.py --port 5000` command.

1. [Microsoft/OLive](https://github.com/microsoft/OLive/blob/8c2bc64c535dd8485bd5bd4cc914b4466fc9c404/olive/web/start_linux.sh#L11) 
This one again runs a script which ultimately uses `python app.py` even in the server mode.

1. [operatorequals/wormnest](https://github.com/operatorequals/wormnest#install---setup---deploy)
Here again, the authors explicitly ask the user to deploy it in production via `python3 app.py`. They even say it's alright if one were to use Apache `mod_rewrite` and expose the app to the internet.

1. [iedadata/usap-dc-website](https://github.com/iedadata/usap-dc-website/blob/master/index.wsgi#L13)
Here again, the developers create a wsgi file to deploy on production but still end up calling the flask development server.

1. [cmusatyalab/opendiamond](https://github.com/cmusatyalab/opendiamond), [cheo0/MercadoEnLineaBack](https://github.com/cheo0/MercadoEnLineaBack),[ChangeWeDer/BaiduWenkuSpider_flaskWeb](https://github.com/ChangeWeDer/BaiduWenkuSpider_flaskWeb) and [sergeKashkin/Simple-RAT](https://github.com/sergeKashkin/Simple-RAT)
These don't come with a guideline on how to deploy to production. So we may assume that they are deployed on a prod grade webserver and have a lower impact as you suggest.

1. [orchest/orchest](https://github.com/orchest/orchest),[ChaoticOnyx/OnyxForum](https://github.com/ChaoticOnyx/OnyxForum) and [clinical-genomics/scout](https://github.com/clinical-genomics/scout)
These do use gunicorn so the severity would be high instead of critical as you say.


As for [ThundeRatz/ThunderDocs](https://github.com/ThundeRatz/ThunderDocs) and [ganga-devs/ganga](https://github.com/ganga-devs/ganga/), fixes for these haven't been merged yet but hey should be soon. But yes, they correctly use a production grade web-server. So the impact is lowered as you say.

So, all in all, 4/13 cases, the apps recommend a user to expose the flask development server to the network. 4/13 don't come with a guideline and 5/13 correctly deploy to gunicorn. This makes the severity critical for 4/13 projects and high for 9/13 projects; assuming the ones which didn't come with a production deployment guideline are correctly deployed via gunicorn.

As for the CVSS vector in the earlier comment https://github.com/github/securitylab/issues/407#issuecomment-976564884, for some reason NIST has changed the impact metrics for my reports. I had incorrectly reported the impact as [AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:H/A:H](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:H/A:H&version=3.0). I had marked high for all fields as can be seen from the original reports which I have listed in the table below. In the absense of any deployment guidelines, I agree the severity of these should be high as you say.

| CVE | Issue |
|:--|:--|
| [CVE-2021-43492](https://nvd.nist.gov/vuln/detail/CVE-2021-43492) | https://github.com/AlquistManager/alquist/issues/42 |
| [CVE-2021-43493](https://nvd.nist.gov/vuln/detail/CVE-2021-43493) | https://github.com/cksgf/ServerManagement/issues/21 |
| [CVE-2021-43494](https://nvd.nist.gov/vuln/detail/CVE-2021-43494) | https://github.com/codingforentrepreneurs/OpenCV-REST-API/issues/2 |
| [CVE-2021-43495](https://nvd.nist.gov/vuln/detail/CVE-2021-43495) | https://github.com/AlquistManager/alquist/issues/43 |
| [CVE-2021-43496](https://nvd.nist.gov/vuln/detail/CVE-2021-43496) | https://github.com/varun-suresh/Clustering/issues/12 |

**Comment by jorgectf** [comment]
@porcupineyhairs 

Thank you for your detailed answer! I see your point and agree that if the project documentation suggests weak deployments, the `Availability Impact` makes sense to be set to `Low` at least, leaving a `Critical` severity.

Looking forward to see how many merged fixes you end up with 🤞

**Comment by porcupineyhairs** [comment]
@jorgectf I have had two more fixes merge through the last week. However, some of the maintainers are not responding to my request to ask for a CVE. So, I will have to apply via MITRE. This may take time. I will update the list with the CVE ids once I have them. I will also post a lgtm run link soon.

**Comment by porcupineyhairs** [comment]
@jorgectf 

I have updated the list to reflect the current state.

[Here's](https://lgtm.com/query/2481481333694945768/) a LGTM link with a run against 13 projects where the vulnerability has been fixed.

**Comment by jorgectf** [comment]
@porcupineyhairs 

What is the current status of the submission?

**Comment by porcupineyhairs** [comment]
@jorgectf 15 Projects have merged their fixes. CVE ID's have been issued for all but two of them. The CVEs aren't publicly disclosed though. I mean they have been issued but the details aren't publicly visible. I contacted MITRE a couple of days back to fix that but they didn't respond.

**Comment by SheuShec** [comment]
I've been noticing that a lot of these issues you're calling out Absolute Path Traversals, but the PoC you're supplying is showing that they are not absolute path traversals. 

The PoC is showing that traversal sequences (f=../../../etc/passwd) are being used rather than absolute paths (f=/etc/passwd).

**Comment by jpcmonster** [comment]
Is the CVE this one?
https://nvd.nist.gov/vuln/detail/CVE-2022-31569
The associated cpe seems to be matching other unrelated jars, I think:
```
jakarta.annotation-api-1.3.5.jar (pkg:maven/jakarta.annotation/jakarta.annotation-api@1.3.5, cpe:2.3:a:oracle:java_se:1.3.5:*:*:*:*:*:*:*, cpe:2.3:a:oracle:projects:1.3.5:*:*:*:*:*:*:*, cpe:2.3:a:projects_project:projects:1.3.5:*:*:*:*:*:*:*) : CVE-2022-31569
javax.jms-api-2.0.jar (pkg:maven/javax.jms/javax.jms-api@2.0, cpe:2.3:a:oracle:projects:2.0:*:*:*:*:*:*:*, cpe:2.3:a:projects_project:projects:2.0:*:*:*:*:*:*:*) : CVE-2022-31569
```

**Comment by SheuShec** [comment]
> @SheuShec I used an automation script to generate those PoC's. Hence, the PoC's could be incorrect. Can you please point me to the code-base you are interested in?
> 
> Those relative paths shown in the PoC may not be always necessary for the exploit to work. The exploit, depending on the situation, should work with absolute path's too.
> 
> But be assured, I have only reported those projects where the path traversal allowed access outside the webroot. Hence, the impact in none of these cases is affected.

Thanks for the response! 

I'm not really interested in a specific code-base per se. I'm more interested in understanding your research. Did you confirm the vulnerabilities in each of these or are they reported based the findings from CodeQL?

**Comment by SheuShec** [comment]
Thanks!

**Comment by jpcmonster** [comment]
hi @porcupineyhairs - I can't be absolutely sure why the CPE is being matched in this case.
I am using the OWASP Dependency Check tool as a gradle plugin in my java project.  
My best _guess_ is that there is something about the CPE associated with the NIST NVD entry that is causing a false positive.
(I have filed an FP report with the Dependency Check project).
But I wanted to post here because there might be something not-quite-right with the cpe.
You should be able to see it on the NIST entry:
```
cpe:2.3:a:projects_project:projects:*:*:*:*:*:*:*:*
```
![image](https://user-images.githubusercontent.com/51028652/179323107-f5640cd8-2eae-4eff-b89b-8cc7e44ef7ac.png)

**Comment by andreasevers** [comment]
I don't know who submitted this CVE, but the CPE identifier matches a boatload of maven dependencies that absolutely should not be matched.

I can only imagine the number of maven builds that fail due to this false positive.

**Comment by JoeNuttall** [comment]
Well, I can tell you that 10 different java projects have failed builds in my department due to this false positive...

Thanks andreasevers and others for making it clear here that it's in error.

**Comment by nagavem** [comment]
Is the suppression for this cve out?

**Comment by cbollmeyer** [comment]
> @jpcmonster I can't see those CPE entries on the webpage you link above. There shouldn't be any java projects affected by any of the CVE's listed above.

This, Sir, is a wrong assumption unfortunately. The OWASP Dependency scanner flags jakarta-annotation-api, JUnit 5 and several other libraries as vulnerable all over the place, all with CVE-2022-31514 (Caoyongqi912/Fan_Platform) and CVE-2022-31569 (RipudamanKaushikDal/projects) so far. It took us half a day to find out it's Python-related and maybe an issue in a number of such projects, but definitely not in the Java libs themselbes. Hopefully there's a way to stop propagation to - as I get it - 90 more such false positives during the next days, each with their own CVE so that you have to write excludes for each and everyone. Maybe better check your script next time.

**Comment by ascopes** [comment]
Can confirm this is affecting builds for me as well. Is there any time frame for this being reverted? Otherwise it will result in having to make dozens of suppressions across anything that validates against https://cve.mitre.org/ or https://nvd.nist.gov/.

**Comment by skavanagh** [comment]
@porcupineyhairs - These projects that are vulnerable to this issue don't seem to have any consumable releases, in those cases there doesn't need to be a CVE open.  Can we use the CPE to target libraries that have consumable releases? Instead of targeting everything under the sun? The need to wildcard everything should tip you off that these aren't actually published libraries and shouldn't have a CVE open. Also, next time it would be great if you published the CVE after the fixes have been merged, so the CPE has a fixed version. It's not great to report a CVE before a maintainer has had a chance to fix and publish it (not relevant in this case b/c there are no consumable releases, but FYI for next time).

**Comment by vdotjansen** [comment]
> @jorgectf I have had two more fixes merge through the last week. However, some of the maintainers are not responding to my request to ask for a CVE. So, I will have to apply via MITRE. This may take time. I will update the list with the CVE ids once I have them. I will also post a lgtm run link soon.

@porcupineyhairs  Did you request the CVE numbers? If you did could you ask them to change the CPE to include the target software python? A lot of non python projects are getting this as a (false positive) effected CVE: https://github.com/jeremylong/DependencyCheck/issues/4671

**Comment by intrigus-lgtm** [comment]
@skavanagh I don't really think `@porcupineyhairs` did anything wrong here.
One *could* argue that some of the CVEs might not be very helpful, because the targeted repositories have very few stars/forks, but there are also a lot of repositories with 100+ stars.

Instead, the problem *seems* to be in `DependencyCheck` itself:
https://github.com/jeremylong/DependencyCheck/issues/4675#issuecomment-1187359752

I'm not really familiar with CPEs, but these two CVEs ([CVE-2021-42055](https://nvd.nist.gov/vuln/detail/CVE-2021-42055), [CVE-2021-42056](https://nvd.nist.gov/vuln/detail/CVE-2021-42056)) have CPEs that are just as broad as [CVE-2022-31569](https://nvd.nist.gov/vuln/detail/CVE-2022-31569), aren't they?

**Comment by skavanagh** [comment]
@intrigus-lgtm I don't know about the other CVEs, but for [CVE-2022-31569](https://nvd.nist.gov/vuln/detail/CVE-2022-31569) the repository it's supposed to be targeting doesn't have any consumable releases. https://github.com/jeremylong/DependencyCheck/issues/4675#issuecomment-1187359752 indicates the problem with requesting a CVE on "random person's personal projects" and it's why the dep-check PR on that thread suppresses [CVE-2022-31569](https://nvd.nist.gov/vuln/detail/CVE-2022-31569) completely.

**Comment by vdotjansen** [comment]
> Instead, the problem _seems_ to be in `DependencyCheck` itself: [jeremylong/DependencyCheck#4675 (comment)](https://github.com/jeremylong/DependencyCheck/issues/4675#issuecomment-1187359752)
>
> I'm not really familiar with CPEs, but these two CVEs ([CVE-2021-42055](https://nvd.nist.gov/vuln/detail/CVE-2021-42055), [CVE-2021-42056](https://nvd.nist.gov/vuln/detail/CVE-2021-42056)) have CPEs that are just as broad as [CVE-2022-31569](https://nvd.nist.gov/vuln/detail/CVE-2022-31569), aren't they?


Yes and no, the dependency checker tries to match a library and if it is not confident it still decides to warn you. It prefers to show false positives over not warning you for false negatives. Part of the CPE is the text project, which means every project that has a project in for example in the URL will trigger as it might be this CVE. 
You are correct those are also not very specific, but this is also, because both CVE's are not for a specific programming language or something like that. Having said that I would still prefer if those are also more specific.

**Comment by skavanagh** [comment]
> Having said that I would still prefer if those are also more specific.

@vdotjansen There is really nothing that can be made more specific, since the CVE references some random piece of [code](https://github.com/RipudamanKaushikDal/projects/issues/21) in a personal repository with no vendor, no version, no release, no package, or no build. There isn't even a license attributed to vulnerable code, so how can anyone use it? [NIST](mailto:nvd@nist.gov) should reconsider [CVE-2022-31569](https://nvd.nist.gov/vuln/detail/CVE-2022-31569), as it's not a consumable library or product. And probably those others as well, they are just not causing as many issues b/c the CPE are more narrow.

**Comment by skavanagh** [comment]
>This blew up more than I imagined.

Well yeah.. it broke some stuff 😆 

> By default, if someone makes a project public, I think it is safe to assume that they believe that the project or parts of it could be reused by someone sometime.

No, it acutally can't be used by anyone if it isn't licenced. [Without granting a licence, only the copyright holder can use the code](https://choosealicense.com/no-permission/). It's why you weren't able to provide a vendor or product name in the CPE. No Licence = No Vendor = No Product.  And I doubt the maintainer would do anything with any of this. AFAICT [CVE-2022-31569](https://nvd.nist.gov/vuln/detail/CVE-2022-31569) was issued for someone's throwaway code.  I don't think this is a edge case that dependency checker is failing to deal with. NIST shouldn't not have approved the CVE, since it isn't assocated with any sort of product.

**Comment by skavanagh** [comment]
> For example, if you publish your source code in a public repository on GitHub, you have accepted the [Terms of Service](https://help.github.com/articles/github-terms-of-service), by which you allow others to view and fork your repository

I can assure you the github terms of service does not licence anyone's unlicensed code.  It allows users to view and fork, and absolves themselves of any responsibility of infringement.  

cpe:2.3:a:RipudamanKaushikDal:projects:*:*:*:*:*:*:*:* doesn't match anything, just like cpe:2.3:a:rexians:rex-web:*:*:*:*:*:*:*:* doesn't match anything. There is nothing to match, since these aren't released, licenced, or consumable by other parties.  

Matching nothing is an improvement over matching everything (but still not much use).  IMO - issuing CVEs for things that aren't consumable by anyone or anything else is not much use and just wasted noise. We will have to agree to disagree (or see what happens)

**Comment by shrimpza** [comment]
This has wasted a good couple of hours of my teams time as we have multiple projects with daily dependency checks going on, and dependency check failures require us to investigate and resolve such issues before we can merge code and produce releases. So we have multiple projects in need of suppressions now, which adds further maintenance burdens down the line.

I understand you may have little/no control over the CPE/mask that is assigned, and that there may be cases where tooling like DependencyCheck causes further issues, but I absolutely believe flagging a bunch of random people's side projects/test code with security vulnerabilities is at best an immense waste of time and resources of everyone involved, and at worst amounts to abuse of these reporting systems.

Where does it stop? Some random person's bash script which has a `rm -rf /` in it because he thought it would be fun to do some testing and happened to commit it to GitHub?

**Comment by ascopes** [comment]
> Matching nothing is an improvement over matching everything (but still not much use). IMO - issuing CVEs for things that aren't consumable by anyone or anything else is not much use and just wasted noise. We will have to agree to disagree (or see what happens)

As much as I appreciate the fact that anyone can open a CVE, I totally agree with this. When creating a CVE, one of the things to consider is the impact of the vulnerability. If this is effectively a personal project that is not being used elsewhere, the impact is minimal. Letting the entire world know via a blanket CVE does not provide reasonable benefit to the development community.

> I understand you may have little/no control over the CPE/mask that is assigned, and that there may be cases where tooling like DependencyCheck causes further issues, but I absolutely believe flagging a bunch of random people's side projects/test code with security vulnerabilities is at best an immense waste of time and resources of everyone involved, and at worst amounts to abuse of these reporting systems.

It appears the CVE has been marked as rejected (https://nvd.nist.gov/vuln/detail/CVE-2022-31569) which is a flag that this CVE was potentially inappropriate. The given reason was:

```
** REJECT ** DO NOT USE THIS CANDIDATE NUMBER. ConsultIDs: none. Reason: This candidate was withdrawn by its CNA. Further investigation showed that no specific affected product had been identified. Notes: none.
```

**Comment by skavanagh** [comment]
> It appears the CVE has been marked as rejected (https://nvd.nist.gov/vuln/detail/CVE-2022-31569) which is a flag that this CVE was potentially inappropriate

Outside of the one that cased the issue, most all of the CVEs here on https://github.com/github/securitylab/issues/669#issuecomment-1117265726 are inappropate for the same reason. I'll be submitting those for re-evalutation as well. What a mess!

**Comment by marcelstoer** [comment]
Instead of bulk-submitting CVEs IMO you should have bulk-submitted GitHub issue for the affected projects.

@skavanagh I also reached out to NIST asking to reject CVE-2022-31514.

**Comment by marcelstoer** [comment]
> The GHSL team along with independent security researchers like me use tools like CodeQL to find security vulnerabilities in open source code. 

👍 

> We try and recognise commonly occuring vulnerable patterns in codebases to prevent them from reoccuring. 

👍 

> In this case, by perforing variant analysis, I found path traversal vulnerabilities in multiple projects. These were reported to their respective maintainers. Those who chose to fix them fixed them, those who didn't didn't. 

👍 

> The entire reporting process was semi-automated and maintainers were given enough time before a disclosure was made. 

👍 

>With each report, a CVE request was made so as to uniquely identify the issue.

👎 Projects that have no license, no vendor, no version, no release, no package, or no build do not deserve to have CVEs IMO. I feel responsible security researchers should understand the - well, responsibility - they have.

> There is no scarcity of record numbers and there's no real harm in registering one.

👎 I disagree, it's just keeping systems and people busy needlessly. In some cases the amount of waste is significant.

> folks who are here blaming me for their failing builds should understand that I am not the reason for their failed builds.

Yes and no. By reporting CVEs that shouldn't exist in the first place you triggered it. Of course you are not responsible for potential bugs in the ODC project ([jeremylong/DependencyCheck!4671](https://github.com/jeremylong/DependencyCheck/issues/4671)). I feel that most people engaged in this discussion here do understand that perfectly well.

**Comment by skavanagh** [comment]
> First, the issue of failing builds, folks who are here blaming me for their failing builds should understand that I am not the reason for their failed builds. The reason behind their failing builds is a bug in the OWASP dependency-check project. Please file a bug report with that project. To repeat, The builds are not failing because I am requesting CVE ids, they fail because the tooling you use is buggy.
 
OWASP dependency-check project is a best effort tool and only as good as the data it utilizes. It breaks up keywords from the vendor, project, and all the fields in the CPE to find matches.  The more accurate the data is for the CVE, the more accurate the tool is (other tools too). We all deal with when these tools fail b/c it flags things it shouldn't, but usually the false-positives are against valid CVEs.  I'm going to have NIST review all your CVEs. You filed junk data to the in NIST database - PEROID!  It would be nice if you realized the mistake an help clean it up, but I guess not 🤷

**Comment by jorgectf** [comment]
Hi all,

Thank you all for bringing this information to our attention. It seems like there was an error assigning the CPEs for some of the CVEs associated with this Bug Slayer submission. We have already conveyed the issue to NIST.

As for the applicability of these CVEs, we do not encourage participants to ask maintainers to request CVEs for small, academic, or personal projects as stated in [our bug bounty rules](https://hackerone.com/github-security-lab):

> Please note that projects which purposely include a vulnerability pattern for testing purposes, projects that are inactive in the last year, and student or academic projects, are considered out of scope.

And therefore, we will not take into account the CVEs assigned to projects matching those conditions.

We are and will be listening to your concerns and suggestions to improve the program and avoid future similar situations.

Thank you.

**Comment by skavanagh** [comment]
I'll appeal to NIST directly on the CVEs in question, but to address the points made

> There has been an assertion of "no license" in a situation where
  multiple GitHub users have been contributing to the codebase. Even if
  you do not know what the license is, the continuing activity by
  multiple persons makes it reasonable to expect that they have arranged
  among themselves to allow use of the software.
 
A lack of licensing would mean a library or project is less likely to be used elsewhere.  A lack of a release is even less likely, A build even more. And on top of all, if it hasn't even been published - I think it's safe to question if anything is really impacted? Do any of these CVEs have any sort of product or service behind them that is vulnerable?

> There might be a bug in https://github.com/jeremylong/DependencyCheck that causes Fan_platform to match ^pkg:maven/org\.junit\.platform/junit\-platform - in other words, the presence of the "platform" substring might be enough for DependencyCheck to guess that there is a dependency'

Dependency check (and other tools) have to do this on purpose. They break up the keywords in the CPE to identify possible vulnerable dependencies. If the NIST data could be used to perfectly identify a vulnerable dependency then there wouldn't be the need to do this and all would be well. This "feature" sometimes accounts for the very problem caused by the negligent submission CVEs. Bad data in = Bad data out. You can't build good tools around bad data. Not saying the NIST data is bad, it's very good! We find legit vulnerabilities all the time, but we need to keep it that way. Submitting CVEs for random github projects does not serve that purpose. The world does not need to know about a  vulnerability in [Uncle Yiba's photo_tag project](https://github.com/uncleYiba/photo_tag) - CVE-2022-31560.  It takes manpower and computing power to ensure things are identified properly. It's not cheap!

**Comment by xcorail** [comment]
Hi @porcupineyhairs,

Thanks for the submission! We have reviewed your report and validated your findings. After internally assessing the findings we have determined this submission is not eligible for a reward under the Bug Bounty program, because the security vulnerabilities have been disclosed publicly, in public Issues and Pull Requests.

The Bug Slayer program requires that you disclose the vulnerabilities following a __coordinated disclosure__ process. Here are the extracts of our rules that mention this:

> To be eligible for a bounty, you must first coordinate disclosure and fix of the vulnerabilities with the maintainers of the projects

> The CVEs must correspond to vulnerabilities that have been disclosed and fixed via coordinated disclosure with the maintainers.

We know that private collaboration between security researchers and maintainers is sometimes difficult, but that is precisely the point of this program. As stated in our rules:

> The goal of this program is to incentivize the collaboration with maintainers to fix vulnerabilities.

Even though this submission was assessed as being ineligible for the bounty program, we appreciate your efforts finding and fixing security vulnerabilities at scale in open source projects, and would like to offer you a thank you reward.

Best regards and happy hacking!

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Closed**.

_For information, the evaluation workflow is the following:
Initial triage > Test run > Results analysis > Query review > Final decision > Pay > Closed_

**Comment by xcorail** [comment]
> I claim, since the vendor for open source projects is typically the owner of the repo and the repository's contributors, by the virtue of me submitting a tested patch, I, for the purposes of CVD, can act as a vendor. @xcorail does not agree with this. They say, the vendor should only be the owner or a previous contributor. I can't deem myself as a vendor and publish a patch.

Indeed I do not agree with you. The reporter cannot be a vendor. If you look at page 17 of the CERT guide that you mention, the vendor is clearly defined, and when they talk about open source they say `Many open source libraries are maintained by a single person or a small independent team; we still refer to these individuals and groups as vendors` which implies that in the open source case, the vendor is a maintainer or a group of maintainers of the projects (and not all contributors). You can totally publish a patch if you want, but you cannot consider yourself as a `vendor` in the documented CVD. The CVD is a **collaboration** between the reporter and the vendor, so the reporter cannot be the vendor.

> By this decision, I believe GHSL is implying that no new alert on LGTM, resulting from a CodeQL stable query, or any additional patch on it, would ever be eligible for Bug Slayer.

The LGTM alerts are indeed visible to the public, but they are not yet verified and triaged. If you use one of these alerts, verify it, create a PoC for it, propose a patch or not, **disclose it privately following the CVD**, and get the consent of the maintainer to create a public PR, then that would be eligible to our program.

**Comment by skavanagh** [comment]
I'll just chime in here too.. b/c you are doing things so wrong.
> I discovered the underlying issue, validated the vulnerability, drafted a report and even created a patch

I don't see how you could have done this. Most of those "projects" don't have a viable application that you could have gotten up and running.  Have you done a reproduction or exploited any of these "applications" that you have alerted the public about? Just b/c something shows in a SAST scan doesn't mean it's vulnerable.  It's got to be explotible in the context of a running application.
> I claim, since the vendor for open source projects is typically the owner of the repo and the repository's contributors, by the virtue of me submitting a tested patch.

Yeah.. that's BS! You couldn't have "tested" the patch on a running "application" there are no applications to run on those CVEs you've opened.  Maybe you've tested the remediation by itself, but you have not in the context of those projects. It's just scraps of code in various repositories.

IMO - Mitre has handled this wrong too. Lots of wrong everywhere.

**Comment by ascopes** [comment]
>  In this case, GHSL deems that by making a public issue and a public PR with a fix, I have violated CVD norms. Instead of filing public issues and public PR's, I should have contacted each maintainer individually and requested them to either fix the vulnerability, provide them a private patch or at least seek prior permission before submitting the PR.

Isn't this the whole definition of responsible disclosure? You don't make dangerous information known to the world until actions have been taken to prevent it being exploited by falling into the wrong hands. By making an issue the center of attention by raising public paperwork for it, you are just holding up a big flashing sign saying "hey, I found an exploit for your application that is dangerous, come see".

> The vulnerability was publicly shown on LGTM well before I attempted to patch them.

LGTM just shows the potential for a vulnerability to be present. It does not understand every edge case that exists to prove something is exploitable.

I have seen applications that LGTM would raise as being vulnerable to CVEs that are just totally incorrect (one being Log4Shell), purely because a dependency has an optional binding to that library.

By raising issues and PRs before a fix has been evaluated and ideally implemented, you are effectively reducing the legwork needed by people scraping for exploits. 

For most Flask apps that are just hobby/pet projects, the impact is somewhat limited, but imagine if you did this for heartbleed, or spectre-like attacks. It would throw everyone into an immediate panic.

**Comment by ascopes** [comment]
>  I sent in a public patch to fix bugs which were valid after manual testing.

That's the point, you explicitly declared and confirmed the vulnerability. Without doing that, it is not clear that the issue was actually a problem in the first place.

Also, by you doing this on your account, you are effectively providing a directory of confirmed exploits to anyone who wishes to misuse that information. As I said, LGTM does not confirm issues. It only flags the potential that they might exist. Any software could be potentially vulnerable but that can be implied by just seeing the dependency list most of the time. LGTM is just linking that and the patterns for it. Neither are disclosing a confirmed issue.

**Comment by xcorail** [comment]
> Basically, I am seeking a few clarifications on GHSL's bounty program. GHSL is rejecting my report saying I am making a public disclosure instead of a CVD. But the thing is if you strictly go by the CVD guidelines, then even LGTM alerts are treated as valid reports. 

I don't agree. LGTM alerts are not `valid reports`. They are just results of a SAST tool, without any human additional analysis of whether the alert is exploitable or not.  

> If GHSL is disqualifying my report due to non-conformance with CVD guidelines, I would like to know if GHSL plans on disqualifing all cases where a public LGTM alert was shown as ineligible on similar grounds.

I already answered this question above. If you transform one of these alerts into an actionable report, by adding your added value as a security researcher (examples from what @ascopes said: `either via clarity, reproductions, confirmation`), and disclose it privately to the maintainer, then it would be eligible.

**Comment by skavanagh** [comment]
I use CodeQL on my projects, BTW!  Love it!
